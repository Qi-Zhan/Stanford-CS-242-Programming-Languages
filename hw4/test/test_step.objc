START VERIFY;

y = z; # Should not step.
x = [l1 = \o.o]; # Should not step.
y = x.l1.l1.l1; # Can take steps.
z = (((x.l1 <- \o.o).l1 <- \o.o).l1 <- \o.o); # Can take steps.

STOP VERIFY;
