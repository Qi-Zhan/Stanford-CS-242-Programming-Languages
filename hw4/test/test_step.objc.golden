y = z;
x = [l1 = \o.(o)];
y = [l1 = \o.(o)];
z = [l1 = \o.(o)];
